scripts/quests/ahtUrhgan/Striking_a_Balance.lua

```diff
@@ -0,0 +1,240 @@
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Saliyahf'] = quest:replaceEvent(687),
+
+            ['Wazyih'] = quest:replaceEvent(689),
+
+            onRegionEnter =
+            {
+                [9] = function(player, region)
+                    return quest:progressEvent(690)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [690] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 2
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Saliyahf'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getLocalVar(player, 'Option') == 0 then
+                        return quest:replaceEvent(692)
+                    else
+                        return quest:replaceEvent(693)
+                    end
+                end,
+            },
+
+            ['Wazyih'] = quest:progressEvent(691),
+
+            onEventFinish =
+            {
+                [691] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [692] = function(player, csid, option, npc)
+                    quest:setLocalVar(player, 'Option', 1)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Saliyahf'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getLocalVar(player, 'Option') == 0 then
+                        return quest:replaceEvent(692)
+                    else
+                        return quest:replaceEvent(693)
+                    end
+                end,
+            },
+
+            ['Wazyih'] = quest:replaceEvent(694),
+
+            onEventFinish =
+            {
+                [692] = function(player, csid, option, npc)
+                    quest:setLocalVar(player, 'Option', 1)
+                end,
+            },
+        },
+
+        [xi.zone.AL_ZAHBI] =
+        {
+            ['550'] =
+            {
+                onTrigger = function(player, npc)
+                    quest:setVar(player, 'Prog', 4)
+                    player:addKeyItem(xi.ki.MUNAHDAS_PACKAGE)
+                    local newPosition = npcUtil.pickNewPosition(npc:getID(), positionTable)
+                    npc:setPos(newPosition.x, newPosition.y, newPosition.z)
+                    return quest:messageSpecial(zones[player:getZoneID()].text.KEYITEM_OBTAINED, xi.ki.MUNAHDAS_PACKAGE)
```

Comment:
This can be simplified to `return quest:keyItem(xi.ki.MUNAHDAS_PACKAGE)`

Reply:
It has the logic for the message included now?

Reply:
Yep, KeyItemAction calls nocUtil.giveKeyItem, which displays message by default